### PR TITLE
chore: setup zksync workflow

### PR DESCRIPTION
**Pull request by nbaztec**
The docs would be updated with the update workflow.

### CODE REVIEW

**Review by elfedy — APPROVED** [review]

**Review by Jrigada — APPROVED** [review]

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -16,9 +16,6 @@ jobs:
         with:
           submodules: recursive
 
-      - name: Install Foundry
-        uses: foundry-rs/foundry-toolchain@v1
```
Comment (Karrq) [inline .github/workflows/update.yml:20]: Maybe we can use https://github.com/bgd-labs/foundry-zksync-toolchain if we need ?

File: book.toml [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
 [output.html]
-git-repository-url = "https://github.com/foundry-rs/book"
-edit-url-template = "https://github.com/foundry-rs/book/edit/master/{path}"
+git-repository-url = "https://github.com/Moonsong-Labs/foundry-zksync-book"
+edit-url-template = "https://github.com/Moonsong-Labs/foundry-zksync-book/edit/main/{path}"
 default-theme = "ayu"
 additional-js = ["src/static/solidity.min.js"]
-cname = "book.getfoundry.sh"
+cname = "96ba0f25.book-3fk.pages.dev"
```
Comment (Karrq) [inline book.toml:19]: could we use github pages for this?

File: scripts/gen_output.sh [hunk 3]
```diff
@@ -3,21 +3,22 @@ set -eo pipefail
...
 need_cmd git
 need_cmd mktemp
 need_cmd sed
 
-need_cmd forge
-need_cmd cast
-need_cmd anvil
-need_cmd chisel
+# download zksync nightlies
+download_nightlies
```
Comment (Karrq) [inline scripts/gen_output.sh:17]: if we use the action above we won't need this, right?

**Review by Karrq — APPROVED** [review]

File: book.toml (hunk 2 above)
Comment (nbaztec) [reply book.toml:19]: It's matter-labs' cloudfare hosting.

File: .github/workflows/update.yml (hunk 1 above)
Comment (nbaztec) [reply .github/workflows/update.yml:20]: I'd rather not rely on it and instead have our own action.

**Pull request merged**